src/Agent/NewRelic/Agent/Core/Attributes/AttributeDefinitionService.cs

```diff
@@ -1112,5 +1118,45 @@ private static string IgnoreEmptyAndWhitespaceErrorGroupValues(string errorGroup
                 .AppliesTo(AttributeDestinations.SpanEvent)
                 .AppliesTo(AttributeDestinations.TransactionTrace)
                 .Build(_attribFilter));
+
+        // new attribute for MessageBrokers - same name as the Externals attribute, but in Agent attributes.
+        // From messaging api spec.
+        private AttributeDefinition<string, string> _brokerServerAddress;
+        public AttributeDefinition<string, string> BrokerServerAddress => _brokerServerAddress ?? (_brokerServerAddress =
+            AttributeDefinitionBuilder.CreateString("server.address", AttributeClassification.AgentAttributes)
+                .AppliesTo(AttributeDestinations.SpanEvent)
+                .AppliesTo(AttributeDestinations.TransactionTrace)
+                .Build(_attribFilter));
+
+        private AttributeDefinition<int, int> _brokerServerPort;
+        public AttributeDefinition<int, int> BrokerServerPort => _brokerServerPort ?? (_brokerServerPort =
+            AttributeDefinitionBuilder.CreateInt("server.port", AttributeClassification.AgentAttributes)
+                .AppliesTo(AttributeDestinations.SpanEvent)
+                .AppliesTo(AttributeDestinations.TransactionTrace)
+                .Build(_attribFilter));
+        
+        private AttributeDefinition<string, string> _messageQueueName;
+        public AttributeDefinition<string, string> MessageQueueName => _messageQueueName ?? (_messageQueueName =
+            AttributeDefinitionBuilder.CreateString("message.queueName", AttributeClassification.AgentAttributes)
+                .AppliesTo(AttributeDestinations.All)
+                .Build(_attribFilter));
+
+        private AttributeDefinition<string, string> _messageRoutingKey;
+        public AttributeDefinition<string, string> MessageRoutingKey => _messageRoutingKey ?? (_messageRoutingKey =
+            AttributeDefinitionBuilder.CreateString("message.routingKey", AttributeClassification.AgentAttributes)
+                .AppliesTo(AttributeDestinations.All)
+                .Build(_attribFilter));
+
+        private AttributeDefinition<string, string> _messagingDestinationRoutingKey;
+        public AttributeDefinition<string, string> MessagingDestinationRoutingKey => _messagingDestinationRoutingKey ?? (_messagingDestinationRoutingKey =
+            AttributeDefinitionBuilder.CreateString("messaging.rabbitmq.destination.routing_key", AttributeClassification.AgentAttributes)
```

Comment:
Is this attribute only ever added for RabbitMq? It might surprise us in the future if we add it for another messaging system and see rabbitmq in the name (the property name does not imply that it's a rabbitmq specific attribute).

Reply:
It is rabbitmq specific, from the messaging spec PR currently in review.  There wasn't anything mentioned about making this for each vendor, however.

I am in favor of making it take a vendor string and using that in the name vs just having this be for rabbitmq.  Thoughts?

Reply:
Either approach works for me. I think we can either just use a RabbitMQ specific property name here, or make the vendor parameterized. Changing the property name might be the simplest. It can be refactored later if something else needs a similarly named attribute.